### PR TITLE
Add example for initializing provider with tf 0.13+

### PR DESCRIPTION
**Pull request by nmuesch**
Add a note about how to declare the provider to use the Terraform Registry when using terraform 0.13 and greater.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -12,6 +12,37 @@ Use the navigation to the left to read about the available resources.
 
 ## Example Usage
 
+Terraform 0.13+ uses the Terraform Registry:
+```hcl
+terraform {
+  required_providers {
+    datadog = {
+      source  = "DataDog/datadog"
+      # version = "~> 1.0"
```
Comment (therve) [inline docs/index.md:21]: Maybe remove that line? I'm afraid people are just going to uncomment it, and I don't really want to list a version here.

File: docs/index.md [hunk 2]
```diff
@@ -12,6 +12,37 @@ Use the navigation to the left to read about the available resources.
...
+    }
+  }
+}
+
+# Configure the Datadog provider
+provider "datadog" {
+  api_key = "${var.datadog_api_key}"
+  app_key = "${var.datadog_app_key}"
```
Comment (therve) [inline docs/index.md:29]: Can you take the chance to use the new syntax without `"${}"` ?

File: docs/index.md [hunk 3]
```diff
@@ -12,6 +12,37 @@ Use the navigation to the left to read about the available resources.
...
+}
+
+# Create a new monitor
+resource "datadog_monitor" "default" {
+  # ...
+}
+
+# Create a new timeboard
+resource "datadog_timeboard" "default" {
```
Comment (therve) [inline docs/index.md:38]: dashboard please! If you can change it below too that'd be great.

**Review by therve — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 1 above)
Comment (nmuesch) [reply docs/index.md:21]: Fair point, I originally left to show an example of where you'd put the version in this setup, but we never showed it the old way either, so removing.

File: docs/index.md (hunk 2 above)
Comment (nmuesch) [reply docs/index.md]: I tried with terraform 0.14, the variables themselves are defined the same so I didn't include it here, but it works to reference them without all the `${}`

**Review by therve — APPROVED** [review]

**Pull request merged**